src/systems/systems/src/overhead/mod.rs

```diff
@@ -448,6 +448,33 @@ impl SimulationElement for MomentaryPushButton {
     }
 }
 
+/// Same implementation as MomentaryPushButton but is only "pressed" for one update even if kept pressed
```

Comment:
i.e. rising edge trigger

Reply:
It is. The actual wasm code would handle both TCA and normal buttons perfectly thanks to this. Problem is that stupid sim code that, when you keep pressing a key, would first send a press, then nothing, then the long press....   Because of that you can't handle both knob and buttons in the same maner :(
It's like they use basic OS key events instead of low level key press events like it's done recently since the 1990's ....

Reply:
Yeah, I found exactly that when looking at your code. Before looking I assumed it would be level trigger rather than edge trigger causing it. 😆

Reply:
@crocket63 idea, write a custom trigger module that define events for expected outputs.

E.g. When you want press and hold events but only want one firing.

Use all three, rising edge, hold, falling edge and examine the results and then fire the events via the custom event handler that could handle any type of events.

The handler would listen to any events (depending on configuration), first registers the press events, await either a press or falling edge. (Maybe with a timer for t amount of times until button has to be released) And then fire only one event accordingly.